### PR TITLE
dotnet: extract file function names

### LINKED ISSUE #952: dotnet: extract function-name features
when a function's name is not obfuscated, extract it (like we do for FLIRT-recognized functions in native binaries)

### PR DESCRIPTION
**Pull request by mike-hunhoff**
fixes #952.

adds file function-name extraction for dotnet files. includes code refactoring and minor fixes for dotnet feature extractor.

### CODE REVIEW

File: capa/features/extractors/dnfile/helpers.py [hunk 1]
```diff
@@ -105,18 +105,24 @@ def get_dotnet_managed_imports(pe: dnfile.dnPE) -> Iterator[Tuple[int, str]]:
             TypeName (index into String heap)
             TypeNamespace (index into String heap)
     """
-    if not hasattr(pe.net.mdtables, "MemberRef") or pe.net.mdtables.MemberRef is None:
+    if not is_dotnet_table_valid(pe, "MemberRef"):
```
Comment (mr-tz) [inline capa/features/extractors/dnfile/helpers.py:108]: good helper!

File: capa/features/extractors/dnfile/helpers.py [hunk 2]
```diff
@@ -167,3 +173,36 @@ def get_dotnet_managed_method_bodies(pe: dnfile.dnPE) -> Iterator[CilMethodBody]
             continue
 
         yield body
+
+
+def is_dotnet_table_valid(pe: dnfile.dnPE, table_name: str) -> bool:
+    return getattr(pe.net.mdtables, table_name, None) is not None
```
Comment (mr-tz) [inline capa/features/extractors/dnfile/helpers.py:179]:
```suggestion
    return bool(getattr(pe.net.mdtables, table_name, None))
```

Comment (mr-tz) [reply capa/features/extractors/dnfile/helpers.py:179]:
```
>>> bool(None)
False
```

File: capa/features/extractors/dnfile/helpers.py [hunk 3]
```diff
@@ -167,3 +173,36 @@ def get_dotnet_managed_method_bodies(pe: dnfile.dnPE) -> Iterator[CilMethodBody]
...
+            name = f"{row.TypeName}::{index.row.Name}"
+
+            # ECMA II.22.37: TypeNamespace can be null or non-null
+            if row.TypeNamespace:
+                # like System.IO.File::OpenRead
+                name = f"{row.TypeNamespace}.{name}"
+
+            token = calculate_dotnet_token_value(index.table.number, index.row_index)
+
+            yield token, name
```
Comment (mr-tz) [inline capa/features/extractors/dnfile/helpers.py:208]: duplicate of above? can these be combined?

Comment (mike-hunhoff) [reply capa/features/extractors/dnfile/helpers.py:208]: agreed there is duplication here; I'd prefer to keep them separate as we are dealing with different metadata tables. however, I'm happy to add some sort of name generation helper for fully qualified method names if there is a strong opinion either way.

Comment (mr-tz) [reply capa/features/extractors/dnfile/helpers.py:208]: not a stopper for me, just noticed the dup

**Review by mr-tz — APPROVED** [review]

**Pull request merged**